application/org.openjdk.jmc.flightrecorder.dependencyview/src/main/java/org/openjdk/jmc/flightrecorder/dependencyview/DependencyView.java

```diff
@@ -0,0 +1,392 @@
+
+		private ModelRebuildRunnable(DependencyView view, IItemCollection items, int packageDepth) {
+			this.view = view;
+			this.items = items;
+			this.packageDepth = packageDepth;
+		}
+
+		private void setInvalid() {
+			this.isInvalid = true;
+		}
+
+		@Override
+		public void run() {
+			view.modelState = ModelState.STARTED;
+			if (isInvalid) {
+				return;
+			}
+			String eventsJson = IItemCollectionJsonSerializer.toJsonString(items);
+			if (isInvalid) {
+				return;
+			} else {
+				view.modelState = ModelState.FINISHED;
+				DisplayToolkit.inDisplayThread().execute(() -> view.setModel(items, eventsJson, packageDepth));
+			}
+		}
+	}
+
+	private enum DiagramType {
+		EDGE_BUNDLING(Messages.getString(Messages.DEPENDENCYVIEW_EDGE_BUNDLING_DIAGRAM), IAction.AS_RADIO_BUTTON, dependencyViewImageDescriptor("edge.png")),
+		CHORD(Messages.getString(Messages.DEPENDENCYVIEW_CHORD_DIAGRAM), IAction.AS_RADIO_BUTTON, dependencyViewImageDescriptor("chord.png"));
+
+		private final String message;
+		private final int action;
+		private final ImageDescriptor imageDescriptor;
+
+		private DiagramType(String message, int action, ImageDescriptor imageDescriptor) {
+			this.message = message;
+			this.action = action;
+			this.imageDescriptor = imageDescriptor;
+		}
+	}
+
+	private class ChangeDiagramTypeAction extends Action {
+		private final DiagramType type;
+
+		ChangeDiagramTypeAction(DiagramType type) {
+			super(type.message, type.action);
+			this.type = type;
+			setToolTipText(type.message);
+			setImageDescriptor(type.imageDescriptor);
+			setChecked(DiagramType.CHORD.equals(type));
+		}
+
+		@Override
+		public void run() {
+			if (this.isChecked()) {
+				diagramType = type;
+				if (currentItems != null) {
+					String eventsJson = IItemCollectionJsonSerializer.toJsonString(currentItems);
+					browser.execute(String.format("updateGraph(`%s`, %d, `%s`);", eventsJson, packageDepth, diagramType.name()));
+				}
+			}
+		}
+	}
+
+	private static ImageDescriptor dependencyViewImageDescriptor(String iconName) {
+		return ResourceLocator.imageDescriptorFromBundle(PLUGIN_ID, DIR_ICONS + iconName).orElse(null); //$NON-NLS-1$
+	}
+
+	private static final int MODEL_EXECUTOR_THREADS_NUMBER = 3;
+	private static final ExecutorService MODEL_EXECUTOR = Executors.newFixedThreadPool(MODEL_EXECUTOR_THREADS_NUMBER,
+			new ThreadFactory() {
+				private ThreadGroup group = new ThreadGroup("DependencyViewCalculationGroup");
+				private AtomicInteger counter = new AtomicInteger();
+
+				@Override
+				public Thread newThread(Runnable r) {
+					Thread t = new Thread(group, r, "DependencyViewCalculation-" + counter.getAndIncrement());
+					t.setDaemon(true);
+					return t;
+				}
+			});
+	private Browser browser;
+	private SashForm container;
+	private IItemCollection currentItems;
+	private volatile ModelState modelState = ModelState.NONE;
+	private ModelRebuildRunnable modelRebuildRunnable;
+	private int packageDepth = 3;
```

Comment:
Would it perhaps be better to start with 2 here? The hierarchical edge bundling gets pretty busy with 3.